tangelo/problem_decomposition/dmet/_helpers/dmet_scf_guess.py

```diff
@@ -41,17 +40,61 @@ def dmet_fragment_guess(t_list, bath_orb, chemical_potential, norb_high, number_
     """
 
     # Construct the fock matrix of the fragment (subtract the chemical potential for consistency)
-    fock_fragment = reduce(np.dot, (bath_orb[:, : norb_high].T, active_fock, bath_orb[:, : norb_high]))
-    norb = t_list[0]
-    if(chemical_potential != 0):
-        for i in range(norb):
-            fock_fragment[i, i] -= chemical_potential
+    fock_fragment = bath_orb[:, : norb_high].T @ active_fock @ bath_orb[:, : norb_high]
+    for i in range(t_list[0]):
+        fock_fragment[i, i] -= chemical_potential
 
     # Diagonalize the fock matrix and get the eigenvectors
     eigenvalues, eigenvectors = scipy.linalg.eigh(fock_fragment)
     eigenvectors = eigenvectors[:, eigenvalues.argsort()]
 
     # Extract the eigenvectors of the occupied orbitals as the guess orbitals
-    frag_guess = np.dot(eigenvectors[ :, : int(number_active_electron/2)], eigenvectors[ :, : int(number_active_electron/2)].T) * 2
+    frag_guess = np.dot(eigenvectors[ :, : int(n_active_electron/2)], eigenvectors[ :, : int(n_active_electron/2)].T) * 2
 
     return frag_guess
+
+
+def dmet_fragment_guess_rohf_uhf(t_list, bath_orb, chemical_potential, norb_high, n_active_electron,
+active_fock_alpha, active_fock_beta, n_active_alpha, n_active_beta):
+    """Construct the guess orbitals.
+
+    Args:
+        t_list (list): Number of [0] fragment & [1] bath orbitals (int).
+        bath_orb (numpy.array): The bath orbitals (float64).
+        chemical_potential (float64): The chemical potential.
+        norb_high (int): The number of orbitals in the fragment calculation.
+        n_active_electron (int): The number of electrons in the fragment
+            calculation.
+        active_fock_alpha (numpy.array): The fock matrix from the low-level
+            calculation for the alpha electrons (float64).
+        active_fock_beta (numpy.array): The fock matrix from the low-level
+            calculation for the beta electrons (float64).
+        n_active_alpha (int): The number octive alpha electrons.
+        n_active_beta (int): The number octive beta electrons.
+
+    Returns:
+        numpy.array, list: The guess orbitals (float64), list of the new
```

Comment:
No, you're returning two objects here, we need two lines for the doc.
Instead of a list for the second one, you may want to return a tuple, as this data-structure is not supposed to grow, and overall packages data not meant to be traversed in any order or something but 2 pieces of co-dependent information.